src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -128,17 +130,20 @@ where
             let len = data.len();
             let sstable_store = self.sstable_store.clone();
             let meta_clone = meta.clone();
+            let policy = self.policy;
             let upload_join_handle = tokio::spawn(async move {
-                sstable_store
-                    .put(
-                        Sstable {
-                            id: table_id,
-                            meta: meta_clone,
-                        },
-                        data,
-                        CachePolicy::Fill,
-                    )
-                    .await
+                if policy == CachePolicy::Fill {
+                    let sst = Sstable::new_with_data(table_id, meta_clone, data.clone())?;
```

Comment:
sstable builder is used in local compaction and compactor compaction. For compactor compaction, I think we may not need to cache the whole sst data in `Sstable`. For local compaction, we can add it to block cache and keep `Sstable` empty like what our current code doing instead of cache the whole data in `Sstable`. 

So in conclusion we may use `Sstable::new(table_id, meta_clone)`